src/features/filter.ts

```diff
@@ -1,12 +1,46 @@
-export const actions = {
-  /* put action creators here */
+/* eslint-disable @typescript-eslint/default-param-last */
+import { Status } from '../types/Status';
+
+type Action = {
+  type: Status;
+  payload: string;
 };
+type Filter = { status: Status; query: string };
+
+const filterActive = (query: string): Action => ({
+  type: 'active',
+  payload: query,
+});
+const filterCompleted = (query: string): Action => ({
+  type: 'completed',
+  payload: query,
+});
+const filterAll = (query: string): Action => ({
+  type: 'all',
+  payload: query,
+});
+
+const applyFilter = (status: Status, query: string): Action => ({
+  type: status,
+  payload: query,
+});
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+const initialFilter: Filter = { status: 'all', query: '' };
+
+const filterReducer = (filter: Filter = initialFilter, action: Action) => {
+  switch (action.type) {
+    case 'active':
+    case 'completed':
+    case 'all':
+      return {
```

Comment:
```suggestion
    case 'CHANGE_FILTER_VALUE':
      return {
```
why do you need three the same actions

Reply:
@etojeDenys Is it correct to use value (which is not in type Status) in case conditions. In my variant, there is no duplicated code and looks more clearer code? Thanks!

Reply:
you can change type description, but no need to use filter types as a action types. action is only 1 - change filter value

then u use values from `filter`